lib/spack/llnl/util/lock.py

```diff
@@ -241,16 +309,21 @@ def acquire_read(self, timeout=None):
         """
         timeout = timeout or self.default_timeout
 
+        locktype = 'READ LOCK'
         if self._reads == 0 and self._writes == 0:
-            self._debug(
-                'READ LOCK: {0.path}[{0._start}:{0._length}] [Acquiring]'
-                .format(self))
+            self._log_acquiring(locktype)
             # can raise LockError.
             wait_time, nattempts = self._lock(fcntl.LOCK_SH, timeout=timeout)
-            self._acquired_debug('READ LOCK', wait_time, nattempts)
             self._reads += 1
+            self._log_acquired(locktype, wait_time, nattempts)
             return True
         else:
+            # Nothing is done with the provided lock timeout so log a warning
+            # if it is different from the one being used.
+            self._log_timeout_warning(timeout)
+
+            # TODO: Should the read count really be incremented if this is a
+            # TODO: a write lock?  Or should there be some other behavior?
```

Comment:
Incrementing `_reads` here is correct -- this tracks nested acquisitions, not actual locks.  We're implementing nesting *on top of* the fcntl locks underneath.  If you "acquire" a read lock inside of a write lock, nothing needs to happen but we need to track how many reads we've nested so we know when to release the write.

Reply:
Okay.  Changed the comment to reflect nested lock tracking.